Player receives email at the end of a Challenge

GabrielaRsanches opened the pull request:
…s condolences or congratulation

Commit:
docs(playerreceivescondolences/congratulations .disl): player receives condolences or congratulation

Commit:
docs(player receives congratulations/condolences.disl): congratulations and condolences use cases ok

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,31 @@
+read_also enums.disl
+read_also ui.disl
+
+ChallengeFailed!
```
You need to place it inside contex > aggregate > service / entity

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,31 @@
+
+ChallengeFailed!
+    rules:
+        - JudgementStatus = Judged
```
= means assignment. Either use == or `is`. Also you are using the enum name as if it were the property.
The correct way to express what you meant is `challenge's judmentStatus is Judged`

GabrielaRsanches replied:
Challenge.@judges JudgementStatus == Judged 
is this right ?

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,31 @@
+        - PlayerOutcome = Failed
+
+    when. 
+        PlayerOutcome is Failed
```
ditto

GabrielaRsanches replied:
did this one like in the above comment
Challenge.@players PlayerOutcome is Failed

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,31 @@
+    when. 
+        PlayerOutcome is Failed
+            NotificationService will send ChallengeFailed! email
+        then.
```
then must come after the condition

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,31 @@
+            NotificationService will send ChallengeFailed! email
+        then.
+            //"Usuário gera imagem do desafio” use case
+            player can share its outcome 
```
this is a UI feature. Not done on the server. We'll need to define it in the UI spec

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,31 @@
+            player can share its outcome 
+        and 
+            // "jogador se desafia" use case
+            player can:
+                -Challenge them self:
+                    create new Challenge
+                    ChallengeYourSelf?
+                // "repeatChallenge" function has to be implemented on api.disl
+                -redo this Challenge:
+                    new Challenge from this same Goal
+                // "juiz desafia jogador" use case
+                -Challenge someone with:
+                    new Challenge 
+                        or
+                    this Goal
+                    ChallengeSomeOne?
```
This is not the responsibility of the NotificationService, neither happens on the API

Commit:
refactor(playerreceivescongratulations/condolences.disl): changes pointed by saulo were made

Commit:
refactor(judgesharesinvitelink.disl): use case is ready to be reviwed

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,13 @@
+read_also enums.disl
+read_also ui.disl
+
+doman/NotificationService calls ChallengeFailed!
```
you need the same structure as in `api.disl`:

```
domain ATQR
  context Challenges
    // your code...
```

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,13 @@
+
+doman/NotificationService calls ChallengeFailed!
+    rules:
+        - JudgementStatus == Judged
```
Please see: https://github.com/tokilabs/atqr/pull/61#discussion_r1054633937

GabrielaRsanches replied:
almost same doubt as what i commented under 

Challenge.@judges.JudgementStatus == Judged

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,13 @@
+        - PlayerOutcome == Failed
+
+    when. 
+        Challenge.PlayerOutcome is Failed
```
Challenge does not have this property (or at least shouldn't). This should com from `Enrollment`

GabrielaRsanches replied:
when. 
        Challenge.@players.PlayerOutcome is Failed
            ChallengeFailed!
            NotificationService calls ChallengeFailed! email
            
            like this ??

GabrielaRsanches replied:
can I say like this, since @players: Enrollment[] ? 

Challenge.@players.PlayerOutcome is Failed

svallory replied:
Almost, `Enrollment`'s property is `outcome`, not `PlayerOutcome` and you forgot the array index. The correct statement is `challenge.@players[0].outcome`. But you could have said the same in natural language: `challenge's  player outcome`

GabrielaRsanches replied:
ok ! thanks

svallory commented on libs/atqr/domain/src/lib/domain-spec/stories/playerReceivesCondolences.disl:
```diff
@@ -0,0 +1,13 @@
+
+    when. 
+        Challenge.PlayerOutcome is Failed
+            NotificationService will send ChallengeFailed! email
```
You just need to fire the `ChallengeFailed!` event. Also, you cannot tell another class to fire an event

Commit:
refactor(playerreceivescongratulations/condolences.disl): pr review

context was implemented, service was pointed and code had proper wrinting changes asked by Saulo

Commit:
feat: just added "player word"

svallory committed:
Final review

Co-authored-by: Gabriela Sanches <GabrielaRsanches@users.noreply.github.com>
Co-authored-by: Raiane Rezende <rairezende@users.noreply.github.com>
